File: Core/JSScript/Core/Utility/List.js
{
if (typeof ALittle === "undefined") ALittle = {};


ALittle.MaxN = function(list) {
	return list.length;
}

ALittle.Push = function(list, object) {
	for (let i = 1; i <= list.length; i += 1) {
		if (list[i - 1] === undefined) {
			list[i - 1] = object;
			return;
		}
	}
	list.push(object);
}

ALittle.Insert = function(list, index, object) {
	list.splice(index - 1, 0, object);
}

ALittle.Remove = function(list, index) {
	list.splice(index - 1, 1);
}

ALittle.Sort = function(list, cmp) {
	list.sort(cmp);
}

}

File: ALittle/Module/Core/Script/Core/Utility/List.lua
-- ALittle Generate Lua
module("ALittle", package.seeall)

local ___thispath = select('1', ...):match("(.+[/\\]).+$") or ""
local ___pairs = pairs
local ___ipairs = ipairs


local insert = table.insert
local remove = table.remove
local maxn = table.maxn
local sort = table.sort
function List_MaxN(list)
	return maxn(list)
end

function List_Push(list, object)
	insert(list, object)
end

function List_Insert(list, index, object)
	insert(list, index, object)
end

function List_Remove(list, index)
	remove(list, index)
end

function List_Sort(list, cmp)
	sort(list, cmp)
end

